### PR TITLE
Krhubert/container

### PR DESCRIPTION
**Pull request by NicolasMahe**

### CODE REVIEW

File: container/network.go [hunk 1]
```diff
@@ -7,23 +7,19 @@ import (
 	docker "github.com/docker/docker/client"
 )
 
-// CreateNetwork creates a Docker Network with a namespace
-func CreateNetwork(namespace []string) (networkID string, err error) {
+// CreateNetwork creates a Docker Network with a namespace.
+func CreateNetwork(namespace []string) (string, error) {
```
Comment (ilgooz) [inline container/network.go]: return with networkID name for docs

File: container/build.go [hunk 2]
```diff
@@ -18,68 +19,64 @@ type BuildResponse struct {
 	Error  string `json:"error"`
 }
 
-// Build a docker image
-func Build(path string) (tag string, err error) {
+// Build builds a docker image.
+func Build(path string) (string, error) {
```
Comment (ilgooz) [inline container/build.go]: use named returns for docs

File: container/network.go [hunk 3]
```diff
@@ -33,36 +29,29 @@ func CreateNetwork(namespace []string) (networkID string, err error) {
...
+// DeleteNetwork deletes a Docker Network associated with a namespace.
+func DeleteNetwork(namespace []string) error {
 	network, err := FindNetwork(namespace)
-	if docker.IsErrNotFound(err) {
-		err = nil
-		return
-	}
-	if err != nil {
-		return
+	if err != nil && !docker.IsErrNotFound(err) {
```
Comment (antho1404) [inline container/network.go]: this is not the same behavior than before
before: if the network is not found we exit with no error
now: if the network is not found we continue and probably have a crash on the `network.ID`

File: container/service.go [hunk 4]
```diff
@@ -10,88 +10,78 @@ import (
...
-func ServiceStatus(namespace []string) (status StatusType, err error) {
-	status = STOPPED
-	_, err = FindService(namespace)
-	if docker.IsErrNotFound(err) {
-		err = nil
-		return
+// ServiceStatus returns the status of the Docker Swarm Servicer.
+func ServiceStatus(namespace []string) (StatusType, error) {
+	if _, err := FindService(namespace); !docker.IsErrNotFound(err) {
+		return STOPPED, err
```
Comment (antho1404) [inline container/service.go]: not the same behavior too. Now if the service is not found we return that it's running which doesn't make sense

File: container/shared_network.go [hunk 5]
```diff
@@ -27,25 +24,23 @@ func createSharedNetworkIfNeeded(client *docker.Client) (err error) {
...
 }
 
-// SharedNetworkID returns the id of the shared network
-func SharedNetworkID() (networkID string, err error) {
+// SharedNetworkID returns the ID of the shared network.
+func SharedNetworkID() (string, error) {
 	client, err := Client()
 	if err != nil {
-		return
+		return "", nil
```
Comment (antho1404) [inline container/shared_network.go]: we need to return the error

File: container/shared_network.go [hunk 6]
```diff
@@ -27,25 +24,23 @@ func createSharedNetworkIfNeeded(client *docker.Client) (err error) {
...
 	}
 	network, err := sharedNetwork(client)
 	if err != nil {
-		return
+		return "", nil
```
Comment (antho1404) [inline container/shared_network.go]: here too

**Comment by antho1404** [comment]
@ilgooz this one was not ready to merge, there is few regressions in the code

**Comment by NicolasMahe** [comment]
I confirm this PR has been reverted. Let's continue the discussion on https://github.com/mesg-foundation/core/pull/336

File: container/network.go (hunk 3 above)
Comment (NicolasMahe) [reply container/network.go]: No, the behavior is the same as before:
The tests pass on this function

File: container/shared_network.go (hunk 5 above)
Comment (NicolasMahe) [reply container/shared_network.go]: Fixed

File: container/service.go (hunk 4 above)
Comment (NicolasMahe) [reply container/service.go]: Fixed

File: container/shared_network.go (hunk 6 above)
Comment (NicolasMahe) [reply container/shared_network.go]: Fixed

File: container/network.go (hunk 3 above)
Comment (NicolasMahe) [reply container/network.go]: Fixed 👍

**Pull request merged**